File: app/src/main/java/com/a60n1/ejashkojme/EjaShkojme.java
package com.a60n1.ejashkojme;

import android.app.Application;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ServerValue;
import com.google.firebase.database.ValueEventListener;
import com.squareup.picasso.OkHttp3Downloader;
import com.squareup.picasso.Picasso;

import net.danlew.android.joda.JodaTimeAndroid;

public class EjaShkojme extends Application {
    private DatabaseReference mUserDatabase;
    private FirebaseAuth mAuth;

    @Override
    public void onCreate() {
        super.onCreate();
        JodaTimeAndroid.init(this);
        FirebaseDatabase.getInstance().setPersistenceEnabled(true);

        Picasso.Builder builder = new Picasso.Builder(this);
        builder.downloader(new OkHttp3Downloader(this, Integer.MAX_VALUE));
        Picasso built = builder.build();
        built.setIndicatorsEnabled(true);
        built.setLoggingEnabled(true);
        Picasso.setSingletonInstance(built);

        mAuth = FirebaseAuth.getInstance();

        if (mAuth.getCurrentUser() != null) {
            mUserDatabase = FirebaseDatabase.getInstance()
                    .getReference().child("users").child(mAuth.getCurrentUser().getUid());
            mUserDatabase.addValueEventListener(new ValueEventListener() {
                @SuppressWarnings("NullableProblems")
                @Override
                public void onDataChange(DataSnapshot dataSnapshot) {
                    if (dataSnapshot != null) {
                        mUserDatabase.child("online").onDisconnect().setValue(ServerValue.TIMESTAMP);
                    }
                }

                @SuppressWarnings("NullableProblems")
                @Override
                public void onCancelled(DatabaseError databaseError) {
                }
            });
        }
    }
}

File: app/src/main/java/com/a60n1/ejashkojme/EjaShkojme.kt
package com.a60n1.ejashkojme

import android.app.Application
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*
import com.squareup.picasso.OkHttp3Downloader
import com.squareup.picasso.Picasso
import net.danlew.android.joda.JodaTimeAndroid

@Suppress("unused")
class EjaShkojme : Application() {
    private var mUserDatabase: DatabaseReference? = null
    private var mAuth: FirebaseAuth? = null
    override fun onCreate() {
        super.onCreate()
        JodaTimeAndroid.init(this)
        FirebaseDatabase.getInstance().setPersistenceEnabled(true)
        val builder = Picasso.Builder(this)
        builder.downloader(OkHttp3Downloader(this, Long.MAX_VALUE))
        val built = builder.build()
        built.setIndicatorsEnabled(true)
        built.isLoggingEnabled = true
        Picasso.setSingletonInstance(built)
        mAuth = FirebaseAuth.getInstance()
        if (mAuth!!.currentUser != null) {
            mUserDatabase = FirebaseDatabase.getInstance()
                    .reference.child("users").child(mAuth!!.currentUser!!.uid)
            mUserDatabase!!.addValueEventListener(object : ValueEventListener {
                override fun onDataChange(dataSnapshot: DataSnapshot) {
                    @Suppress("SENSELESS_COMPARISON")
                    if (dataSnapshot != null)
                        mUserDatabase!!.child("online").onDisconnect().setValue(ServerValue.TIMESTAMP)
                }

                override fun onCancelled(databaseError: DatabaseError) {}
            })
        }
    }
}